ports/servoshell/prefs.rs

```diff
@@ -389,7 +388,16 @@ pub(crate) fn parse_command_line_arguments(args: Vec<String>) -> ArgumentParsing
         process::exit(0);
     };
 
-    let config_dir = opt_match.opt_str("config-dir").map(Into::into);
+    let config_dir = opt_match
+        .opt_str("config-dir")
+        .map(Into::into)
+        .or_else(default_config_dir)
+        .inspect(|path| {
+            if !path.exists() {
+                fs::create_dir_all(path).expect("Failed to create config directory")
```

Comment:
I think we shouldn't panic here, since servo should be able to still run, even without a config dir - logging an error should be sufficient.